commcare_connect/opportunity/tests/test_visit_import.py

```diff
@@ -25,6 +28,30 @@ def test_bulk_update_visit_status():
     assert after_status == {VisitValidationStatus.approved.value}
 
 
+@pytest.mark.django_db
+def test_payment_accrued(opportunity: Opportunity, mobile_user: User):
+    payment_units = PaymentUnitFactory.create_batch(2, opportunity=opportunity)
+    deliver_units = []
+    for payment_unit in payment_units:
+        deliver_units += DeliverUnitFactory.create_batch(2, payment_unit=payment_unit, app=opportunity.deliver_app)
+
+    visits = []
+    for deliver_unit in deliver_units:
+        visits.append(
+            UserVisitFactory(
+                opportunity=opportunity,
+                user=mobile_user,
+                deliver_unit=deliver_unit,
+                entity_id=deliver_unit.payment_unit.name,
+                status=VisitValidationStatus.approved.value,
+            )
+        )
+    update_payment_accrued(opportunity, {mobile_user.id})
+    assert OpportunityAccess.objects.filter(user=mobile_user, opportunity=opportunity).exists()
+    access = OpportunityAccess.objects.get(user=mobile_user, opportunity=opportunity)
+    assert access.payment_accrued == sum(payment_unit.amount for payment_unit in payment_units)
```

Comment:
Nit: it would be nice to have a test that confirmed the user can get a payment unit multiple times (met the criteria for different entities)